examples/Simple.hs

```diff
@@ -61,15 +61,17 @@ instance Monad Exp where
   Lam e    >>= f = Lam (e >>>= f)
   Let bs b >>= f = Let (map (>>>= f) bs) (b >>>= f)
 
-deriveEq1   ''Exp
-deriveOrd1  ''Exp
-deriveRead1 ''Exp
-deriveShow1 ''Exp
-
-instance Eq a => Eq (Exp a) where (==) = eq1
-instance Ord a => Ord (Exp a) where compare = compare1
-instance Show a => Show (Exp a) where showsPrec = showsPrec1
-instance Read a => Read (Exp a) where readsPrec = readsPrec1
+fmap concat $ sequence
+  [ deriveEq1   ''Exp
+  , deriveOrd1  ''Exp
+  , deriveRead1 ''Exp
+  , deriveShow1 ''Exp
+  , [d| instance Eq a => Eq (Exp a) where (==) = eq1
```

Comment:
continuing my previous comment:

```haskell
, deriveEqViaEq1 ''Exp
, deriveOrdViaOrd1 ''Exp
```

?

Reply:
And to answer myself: Deciding the correct context is probably not easy in general, though in these cases it is.

Reply:
There isn't anything in `deriving-compat` for generating `(==) = eq1`, no. `deriving-compat` could offer something like that, I suppose, but most code just derives `Eq` instances directly. Truth be told, I don't fully understand why this code doesn't just derive its own `Eq` instance, but [this comment](https://github.com/ekmett/bound/blob/1278ab30f6faa9812948f903d5ea036c826fd0ba/src/Bound/TH.hs#L116-L117) suggests that there are issues with doing so.

Reply:
Yes, GHC `Eq` deriving doesn't know about `Eq1` so cannot take advantage of that, and derives something else.

Reply:
Err. No. Maybe `deriving instance Eq a => Eq (Exp a)` would be the same, but it would still required `Eq1 Exp` to exist for `Eq (Scope n f a)`

Reply:
... but you'd still need derive them in the same block, so I don't know how much prettier it becomes

Reply:
Right, the all-in-one-TH-splice trick would still be required. On the other hand, you no longer need to import anything from `Data.Functor.Classes`, which saves you some typing.